### DISCUSSION TITLE
(멀티 플레이) 씬이 로드 되고난후 계층뷰에서는 보이고 씬에서는 늦게 보이는 현상

### DISCUSSION THREAD

**Discussion by smilejsu82** [naverz/zepeto-studio-kor#671] [5. Q&A]
월드 1.2.0 입니다 

멀티플레이 테스트중 발생한 상황입니다 

씬이 로드 되고 플레이어가 인스턴스화 되어 계층뷰에서는 보이지만 씬에 늦게 나올때가 있습니다 

케릭터가 씬에 완전히 보였을때를 어떻게 알아낼수 있을까요? 

참고로 OnEnable로는 알아낼수 없습니다 
이미 계층뷰에서는 생성되어 활성화 되었기 때문입니다 

좋은 방법이 없을까요?

**Comment by minjujuu** [comment]
[ClientStarter.ts](https://github.com/naverz/zepeto-multiplay-example/blob/main/Assets/ZepetoScripts/ClientStarter.ts)

플레이어 인스턴스가 씬에 완전히 로드되었을 때 호출되는 다음 함수들을 이용해볼 수 있을 것 같습니다

로컬플레이어만 인식이 필요한 경우,
```
            // [CharacterController] (Local)Player 인스턴스가 Scene에 완전히 로드되었을 때 호출
            ZepetoPlayers.instance.OnAddedLocalPlayer.AddListener(() => {
                const myPlayer = ZepetoPlayers.instance.LocalPlayer.zepetoPlayer;
            });
```
리모트플레이어도 인식이 필요한 경우,
```
           // [CharacterController] Player 인스턴스가 Scene에 완전히 로드되었을 때 호출
            ZepetoPlayers.instance.OnAddedPlayer.AddListener((sessionId: string) => {
                const isLocal = this.room.SessionId === sessionId;
                if (!isLocal) {
                    // 로컬 플레이어가 아닌 경우
                } else {
                    // 로컬 플레이어인 경우
                }
            });
```

**Reply by smilejsu82** [reply]
음 일단 이런 현상입니다...

https://user-images.githubusercontent.com/50188795/165674209-00463d1f-b1a4-4e34-b7db-d5f183670d61.mp4

**Reply by smilejsu82** [reply]
> [ClientStarter.ts](https://github.com/naverz/zepeto-multiplay-example/blob/main/Assets/ZepetoScripts/ClientStarter.ts)
> 
> 플레이어 인스턴스가 씬에 완전히 로드되었을 때 호출되는 다음 함수들을 이용해볼 수 있을 것 같습니다
> 
> 로컬플레이어만 인식이 필요한 경우,
> 
> ```
>             // [CharacterController] (Local)Player 인스턴스가 Scene에 완전히 로드되었을 때 호출
>             ZepetoPlayers.instance.OnAddedLocalPlayer.AddListener(() => {
>                 const myPlayer = ZepetoPlayers.instance.LocalPlayer.zepetoPlayer;
>             });
> ```
> 
> 리모트플레이어도 인식이 필요한 경우,
> 
> ```
>            // [CharacterController] Player 인스턴스가 Scene에 완전히 로드되었을 때 호출
>             ZepetoPlayers.instance.OnAddedPlayer.AddListener((sessionId: string) => {
>                 const isLocal = this.room.SessionId === sessionId;
>                 if (!isLocal) {
>                     // 로컬 플레이어가 아닌 경우
>                 } else {
>                     // 로컬 플레이어인 경우
>                 }
>             });
> ```

감사합니다 ~ 
알려주신 부분을 참고해 해결 했습니다 👍 
그냥 다 들어올때까지 기다렸어요 ㅎ 
어짜피 매칭 되면 시작 하는 월드라 이렇게 해도 될거 같드라구요 
![캡처_2022_04_28_13_14_02_555](https://user-images.githubusercontent.com/50188795/165675336-bad7014d-a0c2-4f2c-b50e-0258ea7ba967.png)

![캡처_2022_04_28_13_12_37_850](https://user-images.githubusercontent.com/50188795/165675156-a2773a29-3c04-4129-bd20-762d467477d0.png)


https://user-images.githubusercontent.com/50188795/165675351-c78e01ce-80f7-456f-b5f6-b7ff32c88b97.mp4

**Reply by smilejsu82** [reply]
각 클라이언트가 준비되었을때 서버에 날려서 체크 한다음 모두 준비가 되면 방장한테 월드 시작하라고 하고 씬전환 하면 될거 같아요 ~ 큰도움이 되었습니다 감사합니다 !

**Reply by minjujuu** [reply]
오 그렇군요, 저도 참고하겠습니다. 공유 감사합니다!
